chaincode/traintuple.go

```diff
@@ -210,20 +210,24 @@ func (traintuple *Traintuple) Save(db *LedgerDB, traintupleKey string) error {
 // -------------------------------------------------------------------------------------------
 
 // createTraintuple adds a Traintuple in the ledger
-func createTraintuple(db *LedgerDB, args []string) (map[string]string, error) {
+func createTraintuple(db *LedgerDB, args []string) (out outputTraintuple, err error) {
 	inp := inputTraintuple{}
-	err := AssetFromJSON(args, &inp)
+	err = AssetFromJSON(args, &inp)
 	if err != nil {
-		return nil, err
+		return
 	}
 
 	key, err := createTraintupleInternal(db, inp, true)
-
 	if err != nil {
-		return nil, err
+		return
 	}
 
-	return map[string]string{"key": key}, nil
+	traintuple, err := db.GetTraintuple(key)
+	if err != nil {
+		return
+	}
+	err = out.Fill(db, traintuple, key)
+	return
```

Comment:
I think you should change the `createTraintupleInternal` so it return the traintuple it just saved. It would prevent one more de-serialization.

Reply:
Not 100% sure as `createTraintupleInternal` is also used inside a compute plan.

Reply:
Sadly for now you right, forget I said anything. 
But in this case I think the `err` should be wrap in an Internal error since its clearly not normal it it doesn't find it. What do you think?